components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -345,16 +359,90 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                         app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
                         app.setTokenRevocationWithIDPSessionTerminationEnabled(
                                 application.isTokenRevocationWithIDPSessionTerminationEnabled());
-                        app.setTokenEndpointAuthMethod(application.getTokenEndpointAuthMethod());
-                        app.setTokenEndpointAuthSignatureAlgorithm(
-                                application.getTokenEndpointAuthSignatureAlgorithm());
-                        app.setSectorIdentifierURI(application.getSectorIdentifierURI());
-                        app.setIdTokenSignatureAlgorithm(application.getIdTokenSignatureAlgorithm());
-                        app.setRequestObjectSignatureAlgorithm(application.getRequestObjectSignatureAlgorithm());
+                        String tokenEndpointAuthMethod = application.getTokenEndpointAuthMethod();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+                            if (enforceFAPIDCR) {
+                                filterAllowedFAPITokenAuthMethods(tokenEndpointAuthMethod);
+                            } else {
+                                filterTokenEndpointAuthMethods(tokenEndpointAuthMethod);
+                            }
+                            app.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
+                        }
+                        String tokenEndpointAuthSigningAlgorithm = application.getTokenEndpointAuthSignatureAlgorithm();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthSigningAlgorithm)) {
+                            if (enforceFAPIDCR) {
+                                filterAllowedFAPISignatureAlgorithms(tokenEndpointAuthSigningAlgorithm);
+                            } else {
+                                filterSignatureAlgorithms(tokenEndpointAuthSigningAlgorithm,
+                                      OAuthConstants.TOKEN_EP_SIGNATURE_ALG_CONFIGURATION);
+                            }
+                            app.setTokenEndpointAuthSignatureAlgorithm(tokenEndpointAuthSigningAlgorithm);
+                        }
+
+                        if (StringUtils.isNotEmpty(application.getSubjectType())) {
```

Comment:
If the subject type is null instead of throwing an exception shall we set the server level default value for this and continue?

Reply:
changed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/fda251e87e1d07a9a8b2debc24038e717fddf9e6